### ISSUE TITLE
createContextualFragment() should do adopt node or some such

### ISSUE THREAD

**Issue by annevk** [w3c/DOM-Parsing#65]
The HTML fragment parsing algorithm creates its own dummy document. Which means that
```
document.createRange().createContextualFragment(`<img src=x onerror=alert(1)>`)
```
should technically not alert as things are currently defined (images are only fetched in active documents). However, it does. In all browsers. And if you check the node document of the returned fragment it will match document, which is an active document.

**Comment by rniwa** [comment]
Yeah, we need to use the same document as that of `Range` object here. I just noticed that [createContextualFragment](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/editing/markup.cpp#L1339) also has code to get rid of html, head, & body elements. That is very strange.
